src/codegen/mod.rs

```diff
@@ -252,6 +252,22 @@ impl ToPtr for proc_macro2::TokenStream {
     }
 }
 
+/// A trait to convert a rust type into an array pointer, optionally const, to the same
+/// type.
+trait ToArrayPtr {
+    fn to_array_ptr(self, is_const: bool, size: usize) -> proc_macro2::TokenStream;
+}
+
+impl ToArrayPtr for proc_macro2::TokenStream {
```

Comment:
I don't think this trait is needed, see below.

Reply:
Fixed